core/sqf/src/seatrans/tm/hbasetmlib2/hbasetm.cpp

```diff
@@ -1218,6 +1220,22 @@ char* HashMapArray::get(int tid)
    return cp_val;
 }
 
+char* HashMapArray::getRegionInfo(int64 tid)
+{
+   jstring js_val = (jstring)(_tlp_jenv->CallObjectMethod(javaObj_, JavaMethods_[JM_GET_REGINFO].methodID, tid));
+   if (getExceptionDetails(NULL)) {
+      tm_log_write(DTM_TM_JNI_ERROR, SQ_LOG_ERR, (char *)"HashMapArray::getRegionInfo()", (char *)_tlp_error_msg->c_str(), -1LL);
+      return NULL;
+   }
+   if(js_val == NULL){
+       return NULL;
+   }
+
+   const jbyte* jb_val = (jbyte*)_tlp_jenv->GetStringUTFChars(js_val, NULL);
```

Comment:
Do we need to worry about memory leaks from C++ references to Java variables? I don't see any code in this module that does worry about this, but the executor code (see for example $TRAF_HOME/../sql/executor/HBaseClient_JNI.cpp) is very careful to push and pop frames to avoid memory leaks. Perhaps the TM is handling this issue in a different way?